airflow/providers/amazon/aws/hooks/redshift.py

```diff
@@ -65,9 +81,9 @@ def cluster_status(self, cluster_identifier: str) -> str:
         """
         try:
             response = self.get_conn().describe_clusters(ClusterIdentifier=cluster_identifier)['Clusters']
-            return response[0]['ClusterStatus'] if response else None
+            return RedshiftClusterStates(response[0]['ClusterStatus']) if response else None
```

Comment:
@o-nikolas had some good thoughts here on using the `Enum`:
https://github.com/apache/airflow/pull/19665#discussion_r753507478

essentially because that's what was used on the EKS operators similar to this `airflow/airflow/providers/amazon/aws/hooks/eks.py`

Reply:
i read the comment and yeah in the 'not found' case that is the focus of the example, sure returning an enum would be fine -- it's completely in your control and a "made-up" status --- but forcing the raw api response into enum is what i feel more iffy about.

and i'd easily say keep the enum there and use it when evaluating the response (e.g. is it paused or available -- use the enum rather than hardcode). just am unsure about converting the raw response in the return value and asking for second opinion :)

Reply:
Good discussion folks!

IMO if there are breaking API changes and it's returning different states than you expect it's going to break your code whether you're using strings or an enum, so it's a moot point. I still vote for using Enums from the start here, add error handling, and if APIs change then this code and other code in the operators will require attention like any other breaking API changes cause.

Reply:
> IMO if there are breaking API changes and the API starts returning different states than you expect it's going to break your code whether you're using strings or an enum, so it's a moot point

Not necessarily.  This hook method is just returning status.  If new statuses appear, it won't break your code, unless you force it into an enum that doesn't know the new status.  And it's far from certain that a new status would break your operator.  If you are waiting for your cluster to become `available` and there's a new status `kindof-almost-available` well that's still not available so your logic is unaffected -- unless of course you are forcing it into an enum and then your code breaks unnecessarily.  That's the scenario that makes me wary of this approach.

Reply:
Hooks are necessarily tied to the API/service they wrap, and I agree with @dstandish  -- we don't want our airflow Hook to suddley _not work at all_ when AWS add a new status code to their API.

Reply:
thanks so much @dstandish and @ashb, should we go back to using the strings and leave `cluster_not_found` to avoid the breaking change?

Reply:
yes please let's avoid breaking

in my view you are still welcome to use the enum for the purpose of evaluating the response from the API  (and also for returning the "not found" `value`) but just don't force the raw response into Enum

thank you 🙏

Reply:
Hey @dstandish,

Yes, you would need to correctly handle the case of a new status, but there's no reason you can't correctly handle this as well as use Eunms (use error handling and coerce anything we haven't seen yet to an UNKOWN enum for example, or just continue to return None like other errors cases can do in this existing code). Certainly each approach has pros and cons, but I see no breaking reason if you chose to use Enums.

Though I'm happy to disagree and commit to using strings if that's what others think as well :)

Reply:
> coerce anything we haven't seen yet to an UNKOWN Enum state, or just continue to return None like other errors cases can do in this existing code

Yup it's true this is an approach that would be more tolerant of the unexpected, it's just not what was done here.

But I would say there is little value in converting in this return value.  If your value conforms to the enum, then your comparisons against the enum would work even without converting in the return.  And if it doesn't conform, then you either fail (current code state), or throw away information (replacing with None or UNKNOWN).  So what's the point of converting it to an enum at all?  I would just leave the raw values alone and use the enums for evaluation.

> Though I'm happy to disagree and commit to using strings if that's what others think as well :)

I don't know that we need to make a universal proclaimation but in this case I think not converting to enum makes sense :)